src/SprykerEco/Service/ComputopApi/Model/Converter/ComputopApiConverter.php

```diff
@@ -0,0 +1,185 @@
+        $this->checkDecryptedResponse($decryptedArray);
+
+        $header = new ComputopApiResponseHeaderTransfer();
+        $header->fromArray($decryptedArray, true);
+        $header->setMId($this->getResponseValue($decryptedArray, ComputopApiConstants::MERCHANT_ID_SHORT));
+        $header->setTransId($this->getResponseValue($decryptedArray, ComputopApiConstants::TRANS_ID));
+        $header->setPayId($this->getResponseValue($decryptedArray, ComputopApiConstants::PAY_ID));
+        //optional
+        $header->setMac($this->getResponseValue($decryptedArray, ComputopApiConstants::MAC));
+        $header->setXId($this->getResponseValue($decryptedArray, ComputopApiConstants::X_ID));
+
+        $header->setIsSuccess(
+            $header->getStatus() === ComputopApiConfig::SUCCESS_STATUS ||
+            $header->getStatus() === ComputopApiConfig::SUCCESS_OK_STATUS
+        );
+        $header->setMethod($method);
+
+        return $header;
+    }
+
+    /**
+     * @param array $responseArray
+     * @param string $key
+     *
+     * @return null|string
+     */
+    public function getResponseValue(array $responseArray, $key)
+    {
+        if (isset($responseArray[$this->formatKey($key)])) {
+            return $responseArray[$this->formatKey($key)];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param string $decryptedString
+     *
+     * @return array
+     */
+    public function getResponseDecryptedArray($decryptedString)
+    {
+        $decryptedArray = [];
+        $decryptedSubArray = explode(self::DATA_SEPARATOR, $decryptedString);
+        foreach ($decryptedSubArray as $value) {
+            $data = explode(self::DATA_SUB_SEPARATOR, $value);
+            $decryptedArray[array_shift($data)] = array_shift($data);
+        }
+
+        return $this->formatResponseArray($decryptedArray);
+    }
+
+    /**
+     * @param array $responseArray
+     *
+     * @throws \SprykerEco\Service\ComputopApi\Exception\ComputopApiConverterException
+     *
+     * @return void
+     */
+    public function checkEncryptedResponse(array $responseArray)
+    {
+        $keys = [
+            ComputopApiConstants::DATA,
+            ComputopApiConstants::LENGTH,
+        ];
+
+        if (!$this->existArrayKeys($keys, $responseArray)) {
+            throw new ComputopApiConverterException(
+                'Response does not have expected values. Please check Computop documentation.'
+            );
+        }
+    }
+
+    /**
+     * @param string $responseMac
+     * @param string $expectedMac
+     * @param string $method
+     *
+     * @throws \SprykerEco\Service\ComputopApi\Exception\ComputopApiConverterException
+     *
+     * @return void
+     */
+    public function checkMacResponse($responseMac, $expectedMac, $method)
+    {
+        if ($this->config->isMacRequired($method) && $responseMac !== $expectedMac) {
+            throw new ComputopApiConverterException('MAC is incorrect');
```

Comment:
I saw `MAC` and `Mac` in the docblocks. Please unify it.

Reply:
If we display here the 2 different MACs, is that a security issue or fine? Is that something that the developer wants to see? Is this visible for developers only?